llvm/lib/IR/RepoTicket.cpp

```diff
@@ -139,12 +139,25 @@ static bool updateDigestUseDependenciesAndContributions(
   GOHash.update('T');
   auto State = AccumulateGODigest(GO, GOHash, GOIMap);
   Changed = Changed || State.Changed;
-  // Update the GO's hash using its contributions and dependencies.
-  auto &GOContributions = State.Contributions;
-  auto &GODependencies = State.Dependencies;
-  for (const GlobalObject *const G : llvm::concat<const GlobalObject *const>(
-           make_range(GOContributions.begin(), GOContributions.end()),
-           make_range(GODependencies.begin(), GODependencies.end()))) {
+  // The hashes of the GO's 'contributions' and 'dependencies' are separately
+  // added to the GO's hash since their respective arrows point are in different
+  // direction. This results in separating the 'visited' table for the two
+  // directions. In other words, the 'contributions' get a visited map and the
+  // 'dependencies' get a separate map. However, 'contributions' cannot form
```

Comment:
I’m not sure that we need these two sentences since they describe an implementation that we don’t have!

Reply:
True, I have removed these two sentences.